src/internal/scheduler/animationFrame.ts

```diff
@@ -50,18 +56,25 @@ export const animationFrame: SchedulerLike = {
          }, delay)
        );
     } else {
-      _queue.push(work, state, subscription);
-      if (_animationFrameId === -1) {
-        _animationFrameId = requestAnimationFrame(flushQueue);
-      }
+      // Start the animation frame loop if it's not already going
+      startAnimationFrameLoop();
+
+      // Push a task to execute onto the queue.
+      const task = { work, state, subscription };
+      _nextQueue.push(task);
+
+      // Set up the teardown
       subscription.add(() => {
-        const index = _queue.indexOf(subscription);
+        // Remove the task from the queue if it's still in there
+        // It won't be in there if it's been executed.
+        const index = _currentQueue.indexOf(task);
         if (index >= 0) {
-          _queue.splice(index - 2, 3);
+          _currentQueue.splice(index, 1);
         }
-        if (_queue.length === 0) {
-          _animationFrameId = -1;
-          cancelAnimationFrame(_animationFrameId);
+
+        // If the current animation frame queue is empty, AND the overall queue is empty, stop the animation frame loop
+        if (_currentQueue.length === 0 && _nextQueue.length === 0) {
```

Comment:
(continuing from the comment above :point_up_2:) if we aren't splicing things out of the queues, we'll probably need to track the following values along with the queues (starting from line 12 above), and update them in `flush()` also:
```ts
let _currentQueue: Task[] = [];
let currentQueueLength = 0;
let currentQueueDisposedCount = 0;
let _nextQueue: Task[] = [];
let nextQueueLength = 0;
let nextQueueDisposedCount = 0;
```

Reply:
I was able to accomplish this with a single counter that is incremented when we schedule, and decremented during unsubscription (teardown).